external_executors/local_cromwell/local_cromwell.py

```diff
@@ -0,0 +1,255 @@
+    Item={{
+        "EventId": event_id,
+        "JobId": "{job_id}",
+        "EventType": "JobLog",
+        "Message": log,
+        "Time": "{{:%Y-%m-%dT%H:%M:%SZ}}".format(datetime.datetime.utcnow())
+    }}
+)
+END
+
+if [ $cromwell_rc == 0 ]; then
+    record_event "JobSucceeded"
+else
+    record_event "JobFailed"
+fi
+
+trap - EXIT
+shutdown -H 5
+"""
+
+# Define boto3 clients
+EC2 = boto3.client('ec2', region_name=REGION)
+
+def workflows_post(event, context):
+    """Handle a POST to /workflows by creating a job.
+
+    The event should contain the following inputs, based off of GA4GH's WES:
+        - workflow_descriptor
+            WDL string that describes the workflow that will be run.
+        - workflow_params
+            JSON string that has the inputs for WDL
+        - workflow_type
+            Ignored, we're only going to try to run WDLs
+        - workflow_type_version
+            Ignored
+        - workflow_dependencies
+            (Optional) base64 encoded ZIP file with WDL dependencies
+    """
+
+    print(event)
+    print(context)
+
+    ec2_profile = os.environ["IAM_PROFILE"]
+    db_table = os.environ["DB_TABLE"]
+
+    # Create a unique id for this job that. We'll pass it to the EC2 instance
+    # that runs the job itself so it can use it to write status updates to the
+    # DB.
+    job_id = str(uuid.uuid4())
+
+    try:
+        workflow_dependencies = event["workflow_dependencies"]
+    except KeyError:
+        workflow_dependencies = ""
+
+    run_script = RUN_SCRIPT.format(
+        inputs_json=shlex.quote(event["workflow_params"]),
+        wdl=shlex.quote(event["workflow_descriptor"]),
+        dependencies=workflow_dependencies,
+        job_id=job_id,
+        db_table=db_table,
+        region=REGION)
+
+    print(run_script)
+    instance = EC2.run_instances(
+        ImageId=AMI,
+        InstanceType=INSTANCE_TYPE,
+        MinCount=1,
+        MaxCount=1,
+        InstanceInitiatedShutdownBehavior='terminate',
+        UserData=run_script,
+        IamInstanceProfile={
+            "Arn": ec2_profile
+        }
+    )
+
+    #instance_id = instance['Instances'][0]['InstanceId']
+
+    return {
+        "statusCode": 200,
+        "headers": {"Content-Type": "application/json"},
+        "body": '{"workflow_id": "' + job_id + '"}'
+        }
+
+def workflow_status(event, context):
+    """Handle GET requests to /workflows/{workflow_id}/status
+
+    Return the status of a given workflow.
+
+    API Gateway is responsible for parsing the URL parameter, so it should
+    show up to this function looking like a POST with a "workflow_id" in
+    the body.
+    """
+    workflow_id = event["workflow_id"]
+    db_table_name = os.environ["DB_TABLE"]
+
+    ddb = boto3.resource("dynamodb", region_name=REGION)
+    table = ddb.Table(db_table_name)
+    response = table.query(
+        KeyConditionExpression=boto3.dynamodb.conditions.Key('JobId').eq(workflow_id)
+    )
+
+    if not response["Count"]:
```

Comment:
What do you expect Count to return to trigger the 404 state?

Reply:
when Count is 0